### PR TITLE
Add motor current tracking application

### PR DESCRIPTION
**Pull request by LoreMoretti**
This PR adds an application which drives the motor in current.

Its main goal is to collect data for friction identification.

cc @isorrentino

### CODE REVIEW

File: utilities/motor-current-tracking/config/robots/ergoCubGazeboV1_1/blf_motor_current_tracking/robot_control.ini [hunk 1]
```diff
@@ -0,0 +1,12 @@
+robot_name                              ergocubSim
+
+# The joints order affects the order of the way points.
+# The last joint is the joint moved first. The first joint is the joint moved last.
+# joints_list                             ("r_hip_pitch", "r_hip_roll", "r_hip_yaw", "r_knee", "r_ankle_pitch", "r_ankle_roll")
+joints_list                             ("r_ankle_roll", "l_ankle_roll")
+
+# remote_control_boards                   ("right_leg")
```
[the remark this answers is not in the document]
Comment (LoreMoretti) [reply utilities/motor-current-tracking/config/robots/ergoCubGazeboV1_1/blf_motor_current_tracking/robot_control.ini]: done in [becc9dc](https://github.com/ami-iit/bipedal-locomotion-framework/pull/894/commits/becc9dca2c23a584c2603a8fc2e4cbf487e0f937)

File: utilities/motor-current-tracking/config/robots/ergoCubGazeboV1_1/blf_motor_current_tracking/robot_control.ini [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+remote_control_boards                   ("left_leg","right_leg")
+positioning_duration                    3.0  #in seconds
+positioning_tolerance                   0.05 #in seconds
+position_direct_max_admissible_error    0.1  #in seconds
```
[the remark this answers is not in the document]
Comment (LoreMoretti) [reply utilities/motor-current-tracking/config/robots/ergoCubGazeboV1_1/blf_motor_current_tracking/robot_control.ini]: done in [becc9dc](https://github.com/ami-iit/bipedal-locomotion-framework/pull/894/commits/becc9dca2c23a584c2603a8fc2e4cbf487e0f937)

File: utilities/motor-current-tracking/config/robots/ergoCubGazeboV1_1/blf_motor_current_tracking/robot_control.ini [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+# The joints order affects the order of the way points.
+# The last joint is the joint moved first. The first joint is the joint moved last.
+# joints_list                             ("r_hip_pitch", "r_hip_roll", "r_hip_yaw", "r_knee", "r_ankle_pitch", "r_ankle_roll")
+joints_list                             ("r_ankle_roll", "l_ankle_roll")
+
+# remote_control_boards                   ("right_leg")
+remote_control_boards                   ("left_leg","right_leg")
+positioning_duration                    3.0  #in seconds
+positioning_tolerance                   0.05 #in seconds
```
[the remark this answers is not in the document]
Comment (LoreMoretti) [reply utilities/motor-current-tracking/config/robots/ergoCubGazeboV1_1/blf_motor_current_tracking/robot_control.ini]: done in [becc9dc](https://github.com/ami-iit/bipedal-locomotion-framework/pull/894/commits/becc9dca2c23a584c2603a8fc2e4cbf487e0f937)

File: utilities/motor-current-tracking/blf-motor-current-tracking.py [hunk 4]
```diff
@@ -0,0 +1,851 @@
...
+
+        while t <= t_end:
+            trajectory.append(motor_current)
+            t = t + dt
+            motor_current = motor_current + dt * speed
+
+        # import matplotlib.pyplot as plt
+        # plt.figure(figsize=(10, 4))  # Create a new figure with a specific size
+        # plt.plot(np.array(trajectory), marker='o')  # Plot the data
+        # plt.show()  # Show the plot
```
[the remark this answers is not in the document]
Comment (LoreMoretti) [reply utilities/motor-current-tracking/blf-motor-current-tracking.py]: done in [becc9dc](https://github.com/ami-iit/bipedal-locomotion-framework/pull/894/commits/becc9dca2c23a584c2603a8fc2e4cbf487e0f937)

File: utilities/motor-current-tracking/blf-motor-current-tracking.py [hunk 5]
```diff
@@ -0,0 +1,851 @@
...
+        "l_knee": 64 * 1e-3,
+        "l_ankle_pitch": 64 * 1e-3,
+        "l_ankle_roll": 177 * 1e-3,
+        "r_hip_roll": 94 * 1e-3,
+        "r_hip_pitch": 64 * 1e-3,
+        "r_hip_yaw": 150 * 1e-3,
+        "r_knee": 64 * 1e-3,
+        "r_ankle_pitch": 64 * 1e-3,
+        "r_ankle_roll": 177 * 1e-3,
+    }
```
[the remark this answers is not in the document]
Comment (LoreMoretti) [reply utilities/motor-current-tracking/blf-motor-current-tracking.py]: done in [29255ac](https://github.com/ami-iit/bipedal-locomotion-framework/pull/894/commits/29255ac3ad53b8abe84ebfa6116e44b98d99d4a1)

File: utilities/motor-current-tracking/blf-motor-current-tracking.py [hunk 6]
```diff
@@ -0,0 +1,851 @@
...
+        "l_knee": 6.0,
+        "l_ankle_pitch": 3.5,
+        "l_ankle_roll": 3.0,
+        "r_hip_roll": 8.0,
+        "r_hip_pitch": 8.0,
+        "r_hip_yaw": 4.0,
+        "r_knee": 6.0,
+        "r_ankle_pitch": 3.5,
+        "r_ankle_roll": 3.0,
+    }
```
[the remark this answers is not in the document]
Comment (LoreMoretti) [reply utilities/motor-current-tracking/blf-motor-current-tracking.py]: done in [29255ac](https://github.com/ami-iit/bipedal-locomotion-framework/pull/894/commits/29255ac3ad53b8abe84ebfa6116e44b98d99d4a1)

File: utilities/motor-current-tracking/blf-motor-current-tracking.py [hunk 7]
```diff
@@ -0,0 +1,851 @@
...
+            "l_ankle_roll",
+            "r_ankle_roll",
+            "l_ankle_pitch",
+            "r_ankle_pitch",
+            "l_hip_yaw",
+            "r_hip_yaw",
+        ]
+        else False
+        for joint in joints_to_control
+    ]
```
[the remark this answers is not in the document]
Comment (LoreMoretti) [reply utilities/motor-current-tracking/blf-motor-current-tracking.py]: done in [67218c7](https://github.com/ami-iit/bipedal-locomotion-framework/pull/894/commits/67218c74cc19eabd37a9c9d42d69f403c78a12d9)

**Comment by LoreMoretti** [comment]
I do not think that the failing `windows-2019@Debug CI` is related to the changes brought by this PR.

**Comment by traversaro** [comment]
> I do not think that the failing `windows-2019@Debug CI` is related to the changes brought by this PR.

The error is:

~~~
2024-09-27T11:08:47.5160396Z LINK : fatal error LNK1201: error writing to program database 'D:\a\bipedal-locomotion-framework\bipedal-locomotion-framework\osqp\build\out\Debug\osqp.pdb'; check for insufficient disk space, invalid path, or insufficient privilege [D:\a\bipedal-locomotion-framework\bipedal-locomotion-framework\osqp\build\osqp.vcxproj]
~~~

**Comment by traversaro** [comment]
> > I do not think that the failing `windows-2019@Debug CI` is related to the changes brought by this PR.
> 
> The error is:
> 
> ```
> 2024-09-27T11:08:47.5160396Z LINK : fatal error LNK1201: error writing to program database 'D:\a\bipedal-locomotion-framework\bipedal-locomotion-framework\osqp\build\out\Debug\osqp.pdb'; check for insufficient disk space, invalid path, or insufficient privilege [D:\a\bipedal-locomotion-framework\bipedal-locomotion-framework\osqp\build\osqp.vcxproj]
> ```

By the way, I usually find this manually, but the new "Explain error" GitHub Copilot feature find it immediately.

**Comment by traversaro** [comment]
Yes, the failure is also on master: https://github.com/ami-iit/bipedal-locomotion-framework/actions/runs/11068539569/job/30754206106 .

**Review by GiulioRomualdi — APPROVED** [review]

**Pull request merged**